test/jdk/java/lang/Thread/UncaughtExceptionsTest.java

```diff
@@ -0,0 +1,206 @@
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static java.lang.System.err;
+import static java.lang.System.out;
+
+/*
+ * @test
+ * @bug 4833089 4992454
+ * @summary Check for proper handling of uncaught exceptions
+ * @author Martin Buchholz
+ * @library /test/lib
+ * @build jdk.test.lib.process.*
+ * @run testng UncaughtExceptionsTest
+ */
+public class UncaughtExceptionsTest {
+
+    @DataProvider
+    public Object[][] testCases() {
+        return new Object[][]{
+            new Object[] { "ThreadIsDeadAfterJoin",
+                           0,
+                           UncaughtExitSimulator.EXPECTED_RESULT,
+                           "Exception in thread \"Thread-0\".*Seppuku"
+            },
+            new Object[] {
+                            "MainThreadAbruptTermination",
+                            1,
+                            UncaughtExitSimulator.EXPECTED_RESULT,
+                            "Exception in thread \"main\".*Seppuku"
+            },
+            new Object[] { "MainThreadNormalTermination", 0, UncaughtExitSimulator.EXPECTED_RESULT, ""},
+            new Object[] { "DefaultUncaughtExceptionHandlerOnMainThread", 1, UncaughtExitSimulator.EXPECTED_RESULT, "" },
+            new Object[] { "DefaultUncaughtExceptionHandlerOnMainThreadOverride", 1, UncaughtExitSimulator.EXPECTED_RESULT, "" },
+            new Object[] { "DefaultUncaughtExceptionHandlerOnNonMainThreadOverride", 0, UncaughtExitSimulator.EXPECTED_RESULT, "" },
+            new Object[] { "DefaultUncaughtExceptionHandlerOnNonMainThread", 0, UncaughtExitSimulator.EXPECTED_RESULT, "" },
+            new Object[] { "ThreadGroupUncaughtExceptionHandlerOnNonMainThread", 0, UncaughtExitSimulator.EXPECTED_RESULT, "" }
+        };
+    }
+
+    @Test(dataProvider = "testCases")
+    public void test(String className, int exitValue, String stdOutMatch, String stdErrMatch) throws Throwable {
+        ProcessBuilder processBuilder = ProcessTools.createJavaProcessBuilder(String.format("Seppuku$%s",className));
```

Comment:
The class renaming looks incomplete. Here and in the error texts above.
Seppuku does not look right as the host class for the test, they are nested classes of UncaughtExceptionsTest.

Reply:
Indeed. Running tests was succeeding because old class files were in place. So running tests is insufficient, one should run them with a pristine directory every time.